File: 1. C, C++/1. Bronze/2920.cpp
#include <iostream>
#include <algorithm>

using namespace std;

int main() {
    // Variables
    int var[8];
    int val[8];

    // Take Input
    for (int i = 0; i < 8; i++) {
        cin >> var[i];
        val[i] = var[i];
    }

    // Sort List
    stable_sort(val, val + 8);

    // Print Result
    if (equal(var, var + 8, val)) cout << "ascending" << endl;
    else {
        reverse(val, val + 8);
        if (equal(var, var + 8, val)) cout << "descending" << endl;
        else cout << "mixed" << endl;
    }
}

File: 2. Java/1. Bronze/2920.java
import java.util.Scanner;
import java.util.Arrays;
import java.util.Collections;

class Main {
    public static void main(String[] args) {
        // Variables
        Scanner sc = new Scanner(System.in);
        Integer var[] = new Integer[8];
        Integer val[] = new Integer[8];

        // Get Input
        for (int i = 0; i < 8; i++) {
            var[i] = sc.nextInt();
            val[i] = var[i];
        }

        // Sort List
        Arrays.sort(val);

        // Print Result
        if (Arrays.equals(var, val)) System.out.println("ascending");
        else {
            Arrays.sort(val, Collections.reverseOrder());
            if (Arrays.equals(var, val)) System.out.println("descending");
            else System.out.println("mixed");
        }
    }
}